iroh-net/src/magicsock.rs

```diff
@@ -408,14 +409,18 @@ impl Inner {
                 } else {
                     trace!(
                         node = %public_key.fmt_short(),
-                        transmit_count = %transmits.len(),
-                        packet_count = &transmits.iter().map(|t| t.segment_size.map(|ss| t.contents.len() / ss).unwrap_or(1)).sum::<usize>(),
-                        len = &transmits.iter().map(|t| t.contents.len()).sum::<usize>(),
+                        transmit_count = %transmits_sent,
                         send_udp = ?udp_addr,
                         send_derp = ?derp_url,
                         "sent transmits"
                     );
-                    Poll::Ready(Ok(n))
+                    assert!(
+                        transmits_sent <= transmits_len,
+                        "too many sent {} > {}",
+                        transmits_sent,
+                        transmits_len
+                    );
```

Comment:
curious why you feel the need for this assert?  It's also the only reason you keep `transmits_len` around.

Reply:
this was from debugging this, probably can remove it